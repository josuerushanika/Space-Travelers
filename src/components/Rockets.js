import { useSelector, useDispatch } from 'react-redux';
import { Button, Badge } from 'react-bootstrap';
import { reserveRocket, cancelRocket } from '../redux/rockets/rocketSlice';
import styling from '../style/Rocket.module.css';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets.rocketListData);
  const dispatch = useDispatch();

  const bookHandler = (id, reserve) => {
    if (reserve) {
      dispatch(cancelRocket(id));
    }
    dispatch(reserveRocket(id));
  };

  return (
    <>
      <div className={styling.maindiv}>
        {rockets.length > 0 && (
          rockets.map((rocket) => (
            <div className={styling.rocketLine} key={rocket.id}>
              <img src={rocket.flickr_image[0]} alt="img" className={styling.img} />
              <div className={styling.rightSide}>
                <h2>{rocket.name}</h2>
                <p>

                  {rocket.reserved && (
                  <Badge
                    bg="primary"
                    className="p-2"
                    style={{ marginRight: '1vw' }}
                  >
                    Reserved
                  </Badge>
                  )}

                  {rocket.description}

                </p>
                <Button
                  type="button"
                  variant={rocket.reserved ? ('outline-secondary') : ('btn btn-primary')}
                  size="md"
                  onClick={() => bookHandler((rocket.id))}
                >
                  {rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket'}
                </Button>
              </div>
            </div>
          ))
        )}
      </div>
    </>
  );
};

export default Rockets;
