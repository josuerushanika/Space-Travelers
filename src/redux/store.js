import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import rocketReducer from './rockets/rocketSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
  },
});

export default store;

