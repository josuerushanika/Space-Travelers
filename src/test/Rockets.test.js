import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Rockets from '../components/Rockets';
import store from '../redux/store';

describe('Rockets', () => {
  it('renders Rockets', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Rockets />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Rockets', () => {
  const mockRockets = [
    {
      id: 'rocket-1',
      name: 'Rocket 1',
      description: 'Description of Rocket 1',
      flickr_image: ['https://www.example.com/image1.jpg'],
      reserved: true,
    },
    {
      id: 'rocket-2',
      name: 'Rocket 2',
      description: 'Description of Rocket 2',
      flickr_image: ['https://www.example.com/image2.jpg'],
      reserved: false,
    },
  ];

  it('should not render the "Reserved" badge when the rocket is not reserved', () => {
    const updatedState = {
      ...store.getState(),
      rockets: {
        ...store.getState().rockets,
        rocketListData: mockRockets,
      },
    };
    store.dispatch({ type: 'SET_STATE', payload: updatedState });

    const { queryByTestId } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    const reservedBadge = queryByTestId('reserved-badge-rocket-2');
    expect(reservedBadge).toBeNull();
  });
});
