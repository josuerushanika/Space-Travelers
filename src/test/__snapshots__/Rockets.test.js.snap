// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets renders Rockets 1`] = `
<div
  className="maindiv"
/>
`;
